### ISSUE TITLE
[Bug] Not in the registry even the Module has been registered

### ISSUE THREAD

**Issue by makecent** [open-mmlab/mmengine#1013]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmengine/issues) and [Discussions](https://github.com/open-mmlab/mmengine/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the latest version(https://github.com/open-mmlab/mmengine).

### Reproduces the problem - code sample

A custom module `DeltaSegmentCoder` is correctly registered.
```python
from mmdet.registry import TASK_UTILS


@TASK_UTILS.register_module()
class DeltaSegmentCoder:
    ......
```
The registered `DeltaSegmentCoder` is then used in another module:
```python
class Something(AnchorHead):
        ......
        from mmdet.registry import TASK_UTILS
        print(TASK_UTILS.module_dict.keys())    # the `DeltaSegmentCoder` is in the print output list (registered)
        super().__init__(
            ...
            bbox_coder=dict(
                     type='DeltaSegmentCoder',
                     target_means=[.0, .0],
                     target_stds=[1.0, 1.0]),
            ....
            **kwargs)
```
In the above `super().__init__(...)`, it builds the class:
```python
from mmdet.registry import TASK_UTILS

class AnchorHead(BaseDenseHead):
    ......
        self.bbox_coder = TASK_UTILS.build(bbox_coder)        # Here raise an error claim that the `DeltaSegmentCoder` was NOT registered
```

### Reproduces the problem - command or script

N/A

### Reproduces the problem - error message

The first row is the print result of `print(TASK_UTILS.module_dict.keys())`. It contains the customized `DeltaSegmentCoder` (the `4-th last`)
```shell
dict_keys(['MaxIoUAssigner', 'ApproxMaxIoUAssigner', 'ATSSAssigner', 'CenterRegionAssigner', 'DynamicSoftLabelAssigner', 'GridAssigner', 'HungarianAssigner', 'BboxOverlaps2D', 'BBoxL1Cost', 'IoUCost', 'ClassificationCost', 'FocalLossCost', 'DiceCost', 'CrossEntropyLossCost', 'MultiInstanceAssigner', 'PointAssigner', 'AnchorGenerator', 'SSDAnchorGenerator', 'LegacyAnchorGenerator', 'LegacySSDAnchorGenerator', 'YOLOAnchorGenerator', 'PointGenerator', 'MlvlPointGenerator', 'RegionAssigner', 'SimOTAAssigner', 'TaskAlignedAssigner', 'UniformAssigner', 'BucketingBBoxCoder', 'DeltaXYWHBBoxCoder', 'DistancePointBBoxCoder', 'LegacyDeltaXYWHBBoxCoder', 'PseudoBBoxCoder', 'TBLRBBoxCoder', 'YOLOBBoxCoder', 'CombinedSampler', 'RandomSampler', 'InstanceBalancedPosSampler', 'IoUBalancedNegSampler', 'MaskPseudoSampler', 'MultiInsRandomSampler', 'OHEMSampler', 'PseudoSampler', 'ScoreHLRSampler', 'DeltaSegmentCoder', 'SegmentOverlaps', 'Anchor1DGenerator', 'MaxIoUAssignerAVA'])

Traceback (most recent call last):
  File "/home/louis/miniconda3/envs/mmengine/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 121, in build_from_cfg
    obj = obj_cls(**args)  # type: ignore
  File "/home/louis/PycharmProjects/ATT/my_modules/models/heads/retina_head_1d.py", line 76, in __init__
    super(RetinaHead1D, self).__init__(
  File "/home/louis/miniconda3/envs/mmengine/lib/python3.8/site-packages/mmdet/models/dense_heads/retina_head.py", line 57, in __init__
    super(RetinaHead, self).__init__(
  File "/home/louis/miniconda3/envs/mmengine/lib/python3.8/site-packages/mmdet/models/dense_heads/anchor_head.py", line 83, in __init__
    self.bbox_coder = TASK_UTILS.build(bbox_coder)
  File "/home/louis/miniconda3/envs/mmengine/lib/python3.8/site-packages/mmengine/registry/registry.py", line 548, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/louis/miniconda3/envs/mmengine/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 100, in build_from_cfg
    raise KeyError(
KeyError: 'DeltaSegmentCoder is not in the task util registry. Please check whether the value of `DeltaSegmentCoder` is correct or it was registered as expected. More details can be found at https://mmengine.readthedocs.io/en/latest/advanced_tutorials/config.html#import-the-custom-module'
```

### Additional information

The problem seems comes from that the imported  `mmdet.TASK_UTILS` will become **empty** when it's called for building modules (TASK_UTILS.build(...)) and the **scope** is NOT `mmdet` but the `mmaction`

**Comment by makecent** [comment]
I find the reason. It's because the logic of cross-library import was changed as the introduce of the `mmengine`.

Previously, `mmlab` repos simply manage their own modules, and we have to register/build a module of a specific repo by importing the specific `repo.Registry`, e.g., `mmdet.MODELS.build(mmdet_model)`, `mmaction.MODELS.build(mmaction_model)`.

Now, the `Registry` in different `mmlab` repos are all managed by the `mmengine`, we just need set a `prefix` to specify a source repo of the module, e.g., `type='mmdet.FocalLoss'`.

Therefore, even though I have registered my customized module `DeltaSegmentCoder` on `mmdet.registry.TASK_UTILS`, the `mmdet.TASK_UTILS.build(type='DeltaSegmentCoder')` will still raise an ERROR when `default_scope = 'mmaction'`because it will search `DeltaSegmentCoder` in the scope of `mmaction.TASK_UTILS`. 

In short, the `scope` of a `Registry` is prior than `where` the `Registry` was from.

I solved my problem by simply registering my `DeltaSegmentCoder` on `mmaction.registry.TASK_UTILS`. Another possible solution is to still register it on `mmdet.registry.TASK_UTILS` but change the config to `dict(type='mmdet.DeltaSegmentCoder')`
